### PR TITLE
Infer token

### PR DESCRIPTION
**Pull request by justAnotherDev**
Use the GitHub provided `${{ github.token }}` as the default so the user does not need to provide one.

Example of `actions/checkout` using the same [syntax](https://github.com/actions/checkout/blob/main/action.yml#L24).

*This update does **NOT** require existing users to change anything.*

**Note:**
This PR also updates the CI file to use whatever the current fork/branch's code is instead of always using master. This allows other branches and forks to test before creating a PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -134,7 +133,6 @@ and removal, and `skipped tests` to detect new skipped and un-skipped tests, as
 See this complete list of configuration options for reference:
 ```yaml
   with:
-    github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (EnricoMi) [inline README.md:137]: This lists all available options, and in some cases you might want to provide a different token than the default one. So this should stay for completeness.

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -38,11 +38,10 @@ jobs:
         python -m pytest --continue-on-collection-errors --junit-xml ../test-results/pytest.xml
 
     - name: Unit Test Results
-      uses: EnricoMi/publish-unit-test-result-action@master
+      uses: ./
```
Comment (EnricoMi) [inline .github/workflows/ci-cd.yml:41]: Getting this to work across branches and forks is great, this might even fix #45.

**Review by EnricoMi — CHANGES_REQUESTED** [review]
Excellent fixes, just one small remark.

File: README.md (hunk 1 above)
Comment (justAnotherDev) [reply README.md:137]: ah, good eye! i updated the name to `your-custom-pat` to try and emphasize that it isn't necessary to provide. happy to switch back to `${{ secrets.GITHUB_TOKEN }}` or whatever other string you want.

https://github.com/EnricoMi/publish-unit-test-result-action/pull/82/commits/05c325652cc07f8398b34deb8f50affd77eaa06a

File: .github/workflows/ci-cd.yml (hunk 2 above)
Comment (justAnotherDev) [reply .github/workflows/ci-cd.yml:41]: yup, likely will 😃

**Review by EnricoMi — APPROVED** [review]
LGTM! Thanks for your contribution!

**Comment by EnricoMi** [comment]
@justAnotherDev can you delete the `.github/.DS_Store` file?

**Comment by justAnotherDev** [comment]
> @justAnotherDev can you delete the `.github/.DS_Store` file?

oops, done! https://github.com/EnricoMi/publish-unit-test-result-action/pull/82/commits/67b0d32239b15b46aaf83e466e17eff13d92b1a1

**Pull request merged**